src/irmin/type/type_pp.ml

```diff
@@ -24,20 +24,57 @@ let t t =
     | Custom c -> c.pp ppf x
     | Map m -> map m ppf x
     | Prim p -> prim p ppf x
-    | _ -> Type_json.pp t ppf x
+    | Var v -> raise (Unbound_type_variable v)
+    | List l -> Fmt.Dump.list (aux l.v) ppf x
+    | Array a -> Fmt.Dump.array (aux a.v) ppf x
+    | Option o -> Fmt.Dump.option (aux o) ppf x
```

Comment:
Will need to implement this by-hand to add parentheses (necessary for e.g. nested `Some`s). The test cases for `result` are doing this properly, and we should be symmetric here.

Reply:
You're right. I've added similar test cases for `option`, and re-implemented option by-hand.
We could be a bit smarter and only parenthesize when needed, but it's probably too much effort for too little reward.